### DISCUSSION TITLE
A workaround to add deadtime to a SISO Transferfunction

### DISCUSSION THREAD

**Discussion by jamestjsp** [ilayn/harold#93] [Show and tell]
The below code can be used to generate a SISO model with deadtime. The only open source package which supports deadtime is the ControlSystems.jl in julia language.

```
import harold
import numpy as np
import matplotlib.pyplot as plt

def deadtime_den(deadtime):
    theta = np.zeros(deadtime)
    for i in range(deadtime):
        if i%2 != 0:
            theta[i] = -0.0
    return theta

G = harold.Transfer([1],[6,2,1])
tss = 40
dt = 1
deadtime = 10
t = np.arange(0,tss,dt)
Gz = harold.discretize(G,dt, 'zoh')
denz = Gz.den.flatten()
deadtimez = int(np.ceil(deadtime/dt))
deadtime_denz = deadtime_den(deadtimez)
Gz.den = np.concatenate((denz, deadtime_denz), axis=None)
harold.step_response_plot(Gz, t)
plt.show()
```

**Comment by ilayn** [comment]
Indeed I'm re-writing the whole transfer function class and we can incorporate this if needed. Delay systems also can be added to the continuous version anyways so that would be a way to unify all this. I'm a bit slow on making it work though. Fingers crossed.

**Reply by jamestjsp** [reply]
That's a good idea and which will attract more users. Adding a Padé function may not be as laborious as adding pure deadtime to the class!
